onnx/defs/controlflow/defs.cc

```diff
@@ -382,16 +388,17 @@ void LoopInferenceFunction(InferenceContext& ctx) {
 
 ONNX_OPERATOR_SET_SCHEMA(
     If,
-    1,
+    11,
     OpSchema()
         .SetDoc("If conditional")
         .Input(0, "cond", "Condition for the if", "B")
         .Output(
             0,
             "outputs",
             "Values that are live-out to the enclosing scope. The return values in "
-            "the `then_branch` and `else_branch` must be of the same shape and same "
-            "data type.",
+            "the `then_branch` and `else_branch` must be of the same same data type. "
+            "If `then_branch` and `else_branch` produce tensors with different "
+            "shapes, conflict shapes would be unknown.",
```

Comment:
It is unclear what this ("conflict shapes would be unknown") means to an user interested in creating a model. Suggest changing this to "Corresponding tensors produced by the then-branch and else-branch are not required to have the same or compatible (static) shapes.". We could talk about how shape-inference will handle it, but I don't think that is part of the operator semantics.

Reply:
Suggestion: "If corresponding outputs from the then-branch and the else-branch have static shapes S1 and S2, then the shape of the corresponding output variable of the if-node (if present) must be compatible with both S1 and S2 as it represents the union of both possible shapes." If we want to add a more detailed explanation, suggest adding to the operator description or IR.md (for the notion of "compatible").

Reply:
Added. Thank you!